test/tests/transfers/multiapp_transfer_transformation/tests

```diff
@@ -10,5 +10,6 @@
     # MultiAppGeometricInterpolationTransfer does not support distributed mesh at this moment
     mesh_mode = 'replicated'
     requirement = "The system shall be able to create virtually translated points during MultiApp transfers."
+    recover = false
```

Comment:
Although this test is using Steady executioner and should be run in recover test, it should not fail. Anyone has insight on this? This is happening when using Steady executioner and try to do recover.

Reply:
I think I know the reason. When we are doing recovery, the multiapp mesh is not uniformly refined even `Mesh/uniform_refine` is greater than zero. Looks like sub-app mesh is not stored in the check point file. I will add an error check on this. I also attempted a fix:

```
diff --git a/framework/src/actions/SetupMeshCompleteAction.C b/framework/src/actions/SetupMeshCompleteAction.C
index 8f27bb4650..dfc60cedcc 100644
--- a/framework/src/actions/SetupMeshCompleteAction.C
+++ b/framework/src/actions/SetupMeshCompleteAction.C
@@ -63,8 +63,10 @@ SetupMeshCompleteAction::act()
      * file based restart and we need uniform refinements, we'll have to postpone
      * those refinements until after the solution has been read in.
      */
-    if (_app.getExodusFileRestart() == false && _app.isRecovering() == false)
+    if (_app.getExodusFileRestart() == false)
     {
+      if (_app.isRecovering() == false || !_app.isUltimateMaster())
+      {
       TIME_SECTION("uniformRefine", 2, "Uniformly Refining");
 
       if (_mesh->uniformRefineLevel())
@@ -86,6 +88,7 @@ SetupMeshCompleteAction::act()
           _displaced_mesh->update();
         }
       }
+      }
     }
   }
   else if (_current_task == "delete_remote_elements_after_late_geometric_ghosting")
```
which does fixed this problem. But I think it is safe to rather issue the error for now.